bfapi/service/user.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2016, RadiantBlue Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy of the
+# License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import logging
+import requests
+import uuid
+
+from datetime import datetime
+from bfapi import db
+from bfapi.config import GEOAXIS_ADDR
+
+TIMEOUT = 24
+
+# TO-DO:
+#   add in audit logging
+#   if receiving auth token directly from ui:
+#       build rest endpoint to receive auth token from ui, then calls geoaxis_token_login and returns user_name + api_key
+#   else
+#       build function to call geoaxis with oauth auth code and derive auth token
+#       build rest endpoint to receive auth code from ui, then calls the above, then calls geoaxis_token_login and returns
+#   plug the user from this in with the user from the user/jobs connection
+
+class User:
+    def __init__(
+            self,
+            *,
+            geoaxis_uid: str,
+            user_name: str,
+            bf_api_key: str = None,
+            created_on: datetime = None ): 
+        self.geoaxis_uid = geoaxis_uid
+        self.user_name = user_name
+        self.bf_api_key = bf_api_key
+        self.created_on = created_on
+
+def geoaxis_token_login(geo_auth_token: str) -> User:
+    userprofile_addr = 'https://{}/ms_oauth/resources/userprofile/me'.format(GEOAXIS_ADDR)
+    try:
+        response = requests.get(
+            userprofile_addr,
+            timeout=TIMEOUT,
+            headers={'Authorization': geo_auth_token},
+        )
+        response.raise_for_status()
+        print(geo_auth_token, response)
+    except requests.ConnectionError:
+        raise CouldNotReachError(userprofile_addr)
+    except requests.HTTPError as err:
+        status_code = err.response.status_code
+        if status_code == 401:
+            raise UnauthorizedError(userprofile_addr)
```

Comment:
Also, it's probably unnecessary to attach the URL here as the only dynamic part of it is the domain (passed in via ENV) and would likely not provide any added benefit during troubleshooting.